src/ConnectedMode/Migration/Wizard/MigrationWizardController.cs

```diff
@@ -43,13 +43,15 @@ internal class MigrationWizardController : IMigrationWizardController
         public event EventHandler MigrationWizardFinished;
 
         public void StartMigrationWizard()
-        { 
-            // TODO show the migration wizard window.
-        }
-
-        private void OnMigrationWizardFinished()
         {
-            MigrationWizardFinished?.Invoke(this, EventArgs.Empty);
+            var migrationWizardWindow = new MigrationWizardWindow();
+
```

Comment:
I can't really test this, I could have the window saved as an internal member variable but that feels a bit ugly. I don't think its okay to leave it untested though so I am not sure how to proceed.

Reply:
We'd have to define an interface to wrap the dialog e.g.

```
    internal interface IMigrationDialog
    {
        DialogResult? ShowModal();
    }
```

... and then inject some way of creating into the controller e.g. by adding a factory interface. We use that approach with e.g. [IAnalysisStatusNotifierFactory](https://github.com/SonarSource/sonarlint-visualstudio/blob/03123f51a5f22aa2bd6e347f00a33ca01e9ee44b/src/CFamily/Analysis/CLangAnalyzer.cs#L58)

Instead of creating a factory interface, we could just add a factory method to the controller class (i.e. a method that just creates and returns a new `IMigrationDialog`), and give tests a way of injecting a different behaviour in the constructor e.g. like we do with [TranslateSpans](https://github.com/SonarSource/sonarlint-visualstudio/blob/03123f51a5f22aa2bd6e347f00a33ca01e9ee44b/src/Integration.Vsix/Analysis/IssueConsumerFactory_IssueHandler.cs#L69) in `IssueHandler`.

However, I wouldn't worry about it for the time being. Let's see if we need to make any more changes first,, and add a note to the board to remind us to look at testing this later.